### ISSUE TITLE
Exception in AttachableCollection<T>.AssociatedObject_Unloaded

### ISSUE THREAD

**Issue by jmclain** [jlaanstra/Windows.UI.Interactivity#4]
Hello jlaanstra,

This is good stuff! However, I have found one problem.

First: I'm using the latest version from the repository with the latest version of Caliburn.Micro on the release version of Windows 8 (WinRT).

In AttachableCollectionOfT.cs the event handler AssociatedObject_Unloaded is called twice for each FrameworkElement when the Popup on which they live goes away. It seems that would have to be WinRT making the call twice - or not; I'm not smart enough about WinRT or W.U.I to be sure.

In any case, an exception is thrown the second time the method tries to remove the event handler because AssociateObject is now null; here's the code I'm using to work around the problem:

Old code:

```
    private void AssociatedObject_Unloaded(object sender, RoutedEventArgs e)
    {
     this.AssociatedObject.Unloaded -= AssociatedObject_Unloaded;
     this.Detach();
    }
```

New code:

```
    private void AssociatedObject_Unloaded(object sender, RoutedEventArgs e)
    {
      if( null != this.AssociatedObject ) {
          this.AssociatedObject.Unloaded -= AssociatedObject_Unloaded;
          this.Detach();
        }
    }
```

**Comment by jlaanstra** [comment]
I'm interested in this bug. Can you please upload a sample solution somewhere which shows this problem? I have not seen this before.

**Comment by jmclain** [comment]
I'll see what I can do; may take a day or so.

**Comment by jmclain** [comment]
Okay, here's a simple example:

  http://pearlsof.com/download/exceptiontest.zip

The solution is fully contained including W.U.I and Caliburn.

Happy hunting ...

**Comment by jlaanstra** [comment]
This issue is fixed in b3375aa6a052e0d938f680db42c20478b2359483.
